### PR TITLE
[BE] Tweak Meta copyright headers

### LINKED ISSUE #90187: Confidential and proprietary notices on several files
### 📚 The doc issue

Hello pytorch team,

I saw confidential and proprietary notices ((c) Facebook, Inc. and its affiliates. Confidential and proprietary.) in following files:
pytorch/android/pytorch_android/src/androidTest/cpp/pytorch_jni_common_test.cpp
pytorch/aten/src/ATen/native/metal/mpscnn/tests/MetalOpTestRunner.h
pytorch/aten/src/ATen/native/metal/mpscnn/tests/MetalOpTestRunner.mm
pytorch/caffe2/utils/knob_patcher.cc
pytorch/caffe2/utils/knobs.cc
pytorch/torch/csrc/distributed/c10d/ParamCommsUtils.cpp
pytorch/torch/csrc/distributed/c10d/quantization/quantization.h
pytorch/torch/csrc/distributed/c10d/quantization/quantization_gpu.h
pytorch/torch/csrc/distributed/c10d/quantization/quantization_utils.h
pytorch/torch/csrc/jit/backends/coreml/cpp/preprocess.cpp
pytorch/torch/csrc/jit/backends/xnnpack/xnnpack_graph_builder.cpp
pytorch/torch/csrc/jit/backends/xnnpack/xnnpack_graph_builder.h
pytorch/torch/csrc/jit/backends/xnnpack/compiler/xnn_compiler.cpp
pytorch/torch/csrc/jit/backends/xnnpack/compiler/xnn_compiler.h
pytorch/torch/csrc/jit/backends/xnnpack/executor/xnn_executor.h
pytorch/torch/csrc/jit/backends/xnnpack/serialization/serializer.cpp
pytorch/torch/csrc/jit/backends/xnnpack/serialization/serializer.h

### Suggest a potential alternative/fix

I am confused by seeing confidential and proprietary notices on GitHub. Would it be possible to either removing these notices?

Thank you for reviewing my issue.

### PR DESCRIPTION
**Pull request by malfet**
s/Facebook, Inc./Meta Platforms, Inc/
s/Confidential and proprietary./This source code is licensed under the BSD-style license/

Per https://www.internalfb.com/intern/wiki/Open_Source/Licenses/Straight_BSD/

Also, add linter that prevents adding those in the future

Fixes https://github.com/pytorch/pytorch/issues/90187

### CODE REVIEW

File: android/pytorch_android/src/androidTest/cpp/pytorch_jni_common_test.cpp [hunk 1]
```diff
@@ -1,4 +1,7 @@
-// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+//
+// This source code is licensed under the MIT license found in the
```
Comment (zpao) [inline android/pytorch_android/src/androidTest/cpp/pytorch_jni_common_test.cpp:3]: PyTorch isn't MIT, so if you're going to have this in any headers, make sure it says BSD. Internal template looks like this:

```suggestion
// This source code is licensed under the BSD-style license found in the
```

File: caffe2/utils/knob_patcher.cc [hunk 2]
```diff
@@ -1,4 +1,7 @@
-// (c) Facebook, Inc. and its affiliates. Confidential and proprietary.
+// (c) Meta Platforms, Inc. and its affiliates.
```
Comment (zpao) [inline caffe2/utils/knob_patcher.cc:1]: I know we're getting nitpicky, but for consistency with the world, our open source headers look slightly different than just removing 'Confidential and proprietary'. You got most of them right, but this and a couple others are wrong.
```suggestion
// // Copyright (c) Meta Platforms, Inc. and affiliates.
```

**Review by zpao — CHANGES_REQUESTED** [review]

**Comment by malfet** [comment]
Thank you for the thorough review. Addressed feedback and updated link to the right license template to use for new files originating from Meta.

**Review by zpao — APPROVED** [review]

**Pull request closed without merging**